src/bootstrap/src/core/config/config.rs

```diff
@@ -2738,12 +2738,17 @@ impl Config {
                 return false;
             }
 
-            self.update_submodule("src/llvm-project");
+            let is_tracked_git_submodule = self.submodules()
+                && GitInfo::new(false, Path::new("src/llvm-project"))
+                    .is_managed_git_subrepository();
```

Comment:
It would be a good idea to unify this logic with the existing checks in `Build::update_existing_submodule`, to avoid the hazard of them accidentally drifting apart.